packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx

```diff
@@ -85,7 +85,7 @@ export const ArrowBlockButton = (props: ArrowBlockButtonProps) => {
             <Icon
               elementDescriptor={leftIconElementDescriptor}
               elementId={leftIconElementId}
-              icon={leftIcon}
+              icon={leftIcon as React.ComponentType}
```

Comment:
So this is a bit complicated 😆 
Due to https://github.com/clerkinc/javascript/blob/caf1f2fd1a8f7b187e152e0e095d01095b41ca98/packages/clerk-js/src/ui/elements/ArrowBlockButton.tsx#L12 this type here is wrong. But you just can't set it to the correct `React.ComponentType` because the type is referenced in other places (as a type alias).

So yeah, instead of fixing stuff in 10 places I just type-casted 🙈

Reply:
What do we need to fix the underlying type issue?
Can we either try to do it now or create a ticket with more details?

Reply:
I don't know yet, since this would require some bigger investigation/changes rather than just keeping things as-is for now.

It comes down to use reusing the props here:

https://github.com/clerkinc/javascript/blob/8434782c53147e4e3333746fd6096212dfcaa51d/packages/clerk-js/src/ui/common/BlockButtons.tsx#L6

Then in other places we use it like this: https://github.com/clerkinc/javascript/blob/8434782c53147e4e3333746fd6096212dfcaa51d/packages/clerk-js/src/ui/components/SignIn/SignInAccountSwitcher.tsx#L51-L57

But the code I changed here only wants `ComponentType`:

https://github.com/clerkinc/javascript/blob/8434782c53147e4e3333746fd6096212dfcaa51d/packages/clerk-js/src/ui/primitives/Icon.tsx#L30

So it's a big problem and fixing this would block me from shipping the more important work

Reply:
The type isn't actually wrong here, see the `isIconElement` boolean on L47 and handling the case where `leftIcon` is an element on L99. It seems that with the TS upgrade though the type narrowing isn't working as it was before 😞 :

![Screenshot 2023-10-16 at 12 18 13 PM](https://github.com/clerkinc/javascript/assets/1289701/81c08fd2-c8da-4cc1-8158-17bc5ac3583e)

I think @LekoArts's fix is good for now, we could also use @ts-expect-error and leave a comment stating we already have a type guard.